nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-cluster/src/main/java/org/apache/nifi/cluster/coordination/heartbeat/ClusterProtocolHeartbeatMonitor.java

```diff
@@ -91,12 +91,8 @@ public ClusterProtocolHeartbeatMonitor(final ClusterCoordinator clusterCoordinat
         this.clusterNodesPath = zkClientConfig.resolvePath("cluster/nodes");
 
         String hostname = properties.getProperty(NiFiProperties.CLUSTER_NODE_ADDRESS);
-        if (hostname == null) {
-            try {
-                hostname = InetAddress.getLocalHost().getHostName();
-            } catch (UnknownHostException e) {
-                throw new RuntimeException("Unable to determine local hostname and the '" + NiFiProperties.CLUSTER_NODE_ADDRESS + "' property is not set");
-            }
+        if (hostname == null || hostname.trim().isEmpty()) {
```

Comment:
For port, we do validate that the port can be parsed via Integer.parseInt -- if empty string, that will fail. I guess the error message may be more clear if we explicitly check for empty string though and indicate that the property is not specified.